R/elasticsearch_parsers.R

```diff
@@ -628,6 +636,26 @@ es_search <- function(es_host
         log_fatal(msg)
     }
     
+    # Other input checks we don't have explicit error messages for
+    .assert(
+        assertthat::is.string(es_host)
+        , es_host != ""
+        , assertthat::is.string(es_index)
+        , es_index != ""
+        , assertthat::is.string(query)
+        , query != ""
+        , assertthat::is.string(scroll)
+        , scroll != ""
+        , assertthat::is.number(max_hits)
+        , max_hits >= 0
+        , assertthat::is.number(n_cores)
+        , n_cores >= 1
```

Comment:
Could you change the n_cores assertions to `assertthat::is.count(n_cores)`? `is.count` checks to make sure it's a single positive integer.

Unrelatedly, I wish `is.count` had a flag or something to allow 0, because checking for a single nonnegative integer is pretty tricky right now.

Reply:
yeah good call. I actually don't mind that they force you to break it into two things. More explicit is better than controlling with flags!